python/ray/data/tests/test_execution_optimizer.py

```diff
@@ -1604,11 +1604,10 @@ def check_transform_fns(op, expected_types):
         assert isinstance(transform_fn, expected_types[i]), transform_fn
 
 
-@pytest.mark.skip("Needs zero-copy optimization for read->map_batches.")
 def test_zero_copy_fusion_eliminate_build_output_blocks(ray_start_regular_shared):
     # Test the EliminateBuildOutputBlocks optimization rule.
     planner = Planner()
-    read_op = get_parquet_read_logical_op()
+    read_op = get_parquet_read_logical_op(parallelism=1)
```

Comment:
any particular reason we need to set `parallelism=1` here?

Reply:
Also, looking at GH history, several other tests were disabled for similar reasons in [this PR](https://github.com/ray-project/ray/commit/cea281c495ef664ddf73bbe05c473122c0255362). The rule was fixed and re-enabled in [this PR](https://github.com/ray-project/ray/pull/38906). Would you mind also re-enabling these tests as well? Thanks!

Reply:
> Also, looking at GH history, several other tests were disabled for similar reasons in [this PR](https://github.com/ray-project/ray/commit/cea281c495ef664ddf73bbe05c473122c0255362). The rule was fixed and re-enabled in https://github.com/ray-project/ray/pull/38906. Would you mind also re-enabling these tests as well? Thanks!


Done - this test is now updated.

> any particular reason we need to set parallelism=1 here?

The current [implementation of OperatorFusionRule._can_fuse](https://github.com/ray-project/ray/blob/master/python/ray/data/_internal/logical/rules/operator_fusion.py#L159) returns false if the additional split factor is greater than 1 which is what would happen if the parallelism is increased beyond 1.